pkg/controller/jobframework/reconciler.go

```diff
@@ -368,7 +369,29 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 	if owner.Name != object.GetName() {
 		return false
 	}
-	return job.EquivalentToWorkload(*wl)
+
+	jobPodSets := job.PodSets()
+
+	if !workload.CanBePartiallyAdmitted(wl) || job.IsSuspended() {
+		// the two sets should fully match.
+		return utilequality.ComparePodSetSlices(jobPodSets, wl.Spec.PodSets, true)
```

Comment:
I'm still against this simple comparison, because it might not work for a Job that fails in between.

Reply:
In which case?

Reply:
Also the scope of the PR is just to move the common functionality from individual the concrete implementations into the framework. I believe it dose that without any regressions. We can open followup tickets to rework if necessary.

Reply:
Yes, that is true, this doesn't seem to introduce regressions.
However, I'm working on the e2e test with preemption. If we happen to find a bug, merging this PR might make it harder to cherry-pick a fix into the previous release.

Reply:
The same is for any other code changes that go only on one branch.

Reply:
Let me get the E2E test first to see how common the problem is.